tests/chainer_tests/links_tests/connection_tests/test_lstm.py

```diff
@@ -184,11 +184,14 @@ def test_to_cpu_gpu(self):
         self.check_to_cpu(self.link.h)
 
     def check_to_cpu_to_gpu(self, s):
-        self.link.to_gpu()
+        # If `self.link` already resides on the GPU, the first `to_gpu` calls
+        # do nothing and the first device transfer occurs at the first
+        # `to_cpu`. We must therefore pass `allow_unchain=True` there.
+        self.link.to_gpu(allow_unchain=True)
```

Comment:
This is not necessarily a GPU-GPU transfer https://github.com/chainer/chainer/pull/7507/files#diff-c27c7baca59d10196efd36f373e887fbR202

Reply:
Could you give more explanation?
I don't understand the logical association between the two sentences.

Reply:
Btw, the link does not seem to work.

Reply:
Updated the link.

Reply:
Do you mean the two sentences in the comment in the code above this line? If yes, when this line is executed
- if the link already resides on the GPU, it essentially does nothing, not even unchain. So, we must again when transferring to CPU further down this check method, again pass `allow_unchain=True` to avoid the deprecation error
- if on the other hand the link resides on the CPU, then obviously this list must look like this in order to avoid the deprecation error.

Reply:
Let's fix so that `allow_unchain=True` takes effect in GPU-GPU case.